src/play_sound_event.rs

```diff
@@ -1,5 +1,15 @@
-use bevy::prelude::Event;
+use libfmod::Studio;
 
-// TODO: Replace this with an EventDescription?
-#[derive(Event)]
-pub struct PlaySoundEvent(pub &'static str);
+pub struct OneShotPlayer {
+    pub studio: Studio,
```

Comment:
There's already a `NonSend<Studio>` resource. Maybe try moving this logic into a system?

Reply:
Hmm I'm not sure how we would do that. Unless we make the user spawn an entity with a OneShotPlayer component for example